### DISCUSSION TITLE
'int' object has no attribute 'value' on sign_pdf

### DISCUSSION THREAD

**Discussion by ilpadrinohack** [MatthiasValvekens/pyHanko#248] [Q&A]
Hi Matthias and everybody:

With the update of pyHanko from 0.13 o the latest one, I have a problem when trying to sign a pdf document. This is my code:

         `w = IncrementalPdfFileWriter(BytesIO(MINIMAL))
          fname = fname.replace('.pdf', '-f.pdf')
                    with open(fname, 'wb') as outf:
                        
                        out = signers.PdfSigner(signature_meta=signers.PdfSignatureMetadata(
                                                        field_name='Sigdni',
                                                        reason='Firma',
                                                        location='',
                                                        use_pades_lta=True,
                                                        subfilter=fields.SigSeedSubFilter.PADES,
                                                        embed_validation_info=True,
                                                        validation_context=vc
                                                        ),
                                        signer=signerp,
                                        timestamper=None,
                                        stamp_style=style,
                                        new_field_spec=sp
                                        ).sign_pdf(w,output=outf)`

 And the error is _"'int' object has no attribute 'value'"_
I think (not sure) that the problem is on the _sign_pdf_ function.

If I change my code calling in a different way to _IncrementaPdfFileWriter_ function as follow:

        `fname = fname.replace('.pdf', '-f.pdf')
                    with open(fname, 'wb') as outf:
                        w = IncrementalPdfFileWriter(outf)
                        
                        out = signers.PdfSigner(signature_meta=signers.PdfSignatureMetadata(
                                                        field_name='Sigdni',
                                                        reason='Firma',
                                                        location='',
                                                        use_pades_lta=True,
                                                        subfilter=fields.SigSeedSubFilter.PADES,
                                                        embed_validation_info=True,
                                                        validation_context=vc
                                                        ),
                                        signer=signerp,
                                        timestamper=None,
                                        stamp_style=style,
                                        new_field_spec=sp
                                        ).sign_pdf(w)`

I get the error _"Illegal PDF header"_
I have been checking the documentation and haven't been able to notice any change in the implementation in the sign_pdf call.

Any help will be appreciated.
Thanks in advance

**Comment by MatthiasValvekens** [comment]
Hi @ilpadrinohack,

The fact that your second example doesn't work is to be expected: the stream passed into the `IncrementalPdfFileWriter` needs to contain the "base" file. You're invoking it on an empty stream.

As for the first example: please post the full stack trace. It's somewhat likely that the error you're seeing was present all along, but surfaced only now due to various deep changes in the validation code. Whether it's a data correctness issue or a bug is impossible to tell without a stack trace.

EDIT: moving this to a discussion. Will create an issue if/when we identify a concrete bug.

**Comment by ilpadrinohack** [comment]
Thanks Matthias:
This is the stack trace:


           `Traceback (most recent call last):
            File "/root/Descargas/programacion/firma_digital/pades_sign/ff/firma.py", line 279, in test_simple_sign
    out = signers.PdfSigner(signature_meta=signers.PdfSignatureMetadata(
           File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/signers/pdf_signer.py", line 1408, in sign_pdf
    result = asyncio.run(
           File "/usr/lib64/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
           File "/usr/lib64/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
           File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/signers/pdf_signer.py", line 1466, in async_sign_pdf
    signing_session = self.init_signing_session(
          File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/signers/pdf_signer.py", line 1152, in init_signing_session
    sig_field_ref = next(cms_writer)
          File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/signers/cms_embedder.py", line 453, in write_cms
    field_created, sig_field_ref = _get_or_create_sigfield(
          File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/signers/cms_embedder.py", line 131, in _get_or_create_sigfield
    apply_sig_field_spec_properties(
          File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/fields.py", line 1701, in apply_sig_field_spec_properties
    sig_field_spec.seed_value_dict.as_pdf_object()
          File "/usr/local/lib/python3.10/site-packages/pyhanko/sign/fields.py", line 854, in as_pdf_object
    pdf_name('/Ff'): generic.NumberObject(self.flags.value),`
          AttributeError: 'int' object has no attribute 'value'
------------------------------------------------------------
         'int' object has no attribute 'value'`

Hope this can help

Thanks

**Comment by ilpadrinohack** [comment]
SOLVED
I don't know why but if I use:

     `sv = fields.SigSeedValueSpec(
                        reasons=[],            
                        flags=3,
                        )`
And after  `sp = fields.SigFieldSpec('Sigdni', seed_value_dict=sv, on_page=pagina_firma-1, box=(x1, y1, x2, y2),)`

I get the error mentioned above.

 Removing "seed_value_dict" from SigFieldSpec as follow:
   `sp = fields.SigFieldSpec('Sigdigp',  on_page=pagina_firma-1, box=(x1, y1, x2, y2),)`

works perfectly.

Thanks Matthias

**Reply by MatthiasValvekens** [reply]
Yeah, I should probably redo some needlessly general examples in the docs to no longer involve seed values, since they're a gimmick that 99.9% of users won't ever need.

For reference, the issue was that you needed to specify `SigSeedValFlags(3)` instead of `3`. Weird that that didn't trigger a warning in your IDE, though.
